pixelator.html

```diff
@@ -1,126 +1,172 @@
-      <input type="number" id="image-multiplier" min="0" max="666" value="5" />
-      <button id="btn">Update</button>
-    </div>
-    <img src="veera.png" />
-    <script type="text/javascript">
-      // Are you KIDDING me with this type casting?
-
-      const canvas = /** @type {HTMLCanvasElement} */ document.querySelector('#target');
-      const btn = document.querySelector("#btn");
-      const in_file = document.querySelector("#image-file");
-      const in_nbr = document.querySelector("#image-multiplier");
-
-      let pixelMultiplier = parseInt(in_nbr.value);
-
-      canvas.width = window.innerWidth;
-      canvas.height = window.innerHeight;
-
-      const ctx = /** @type {CanvasRenderingContext2d} */ canvas.getContext('2d');
-
-      const img = document.querySelector('img');
-
-      img.addEventListener('load', () => {
-        let imageHeight = img.height;
-        let imageWidth = img.width;
-
-        // draw centered image
-        const posX = Math.ceil(canvas.width/2 - imageWidth/2);
-        const posY = Math.ceil(canvas.height/2 - imageHeight/2);
-        ctx.drawImage(img, posX, posY, imageWidth, imageHeight);
-
-        // read rendered image pixels
-        const imageData = ctx.getImageData(posX, posY, imageWidth, imageHeight);
-        const pixelvalues = imageData.data;
-        // console.log(pixelvalues)
-
-        ctx.clearRect(0, 0, canvas.width, canvas.height);
-
-        // lööp over image, each 5th pixel
-        let loopkiller = 100000; // around 100k
-
-        for (let row = 1; row < imageHeight; row = row + pixelMultiplier) {
-          for (let col = 1; col < imageWidth; col = col + pixelMultiplier) {
-            loopkiller--;
-            if (loopkiller < 0) { return console.log("killed"); }
-            
-            // get 1st pixel color
-            const pixelReference = ((row*imageWidth)+(col))*4;
-            if (pixelReference === null || typeof pixelReference === "undefined") {return console.log("Invalid pixel reference"); }
-
-            const color = `rgb(
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <meta name="author" content="Joel Peltonen">
+  <title>Document</title>
+  <style>
+    * {
+      margin: 0;
+      padding: 0;
+    }
+
+    body {
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+      align-items: center;
+      width: 100vw;
+      height: 100vh;
+    }
+
+    #controls {
+      display: flex;
+      flex-direction: column;
+      justify-self: center;
+      text-align: right;
+      background-color: whitesmoke;
+    }
+
+    .input-group {
+      display: flex;
+      align-items: center;
+      margin: .5em;
+    }
+
+    label {
+      width: 10em;
+      margin-right: .5em;
+    }
+
+    #graphics {
+      display: flex;
+      justify-content: center;
+      align-items: center;
```

Comment:
Just curious, did you test this on mobile? My original is quite bad on mobile, but it was testable and there was no scroll

Reply:
As I mentioned, no;-)